python/tvm/driver/tvmc/runner.py

```diff
@@ -0,0 +1,450 @@
+
+            logger.debug(
+                "generating data for input '%s' (shape: %s, dtype: %s), using fill-mode '%s'",
+                input_name,
+                shape,
+                dtype,
+                fill_mode,
+            )
+            data = generate_tensor_data(shape, dtype, fill_mode)
+            inputs_dict[input_name] = data
+
+    return inputs_dict
+
+
+def run_module(
+    module_file,
+    hostname,
+    port=9090,
+    rpc_key=None,
+    device=None,
+    inputs=None,
+    fill_mode="zeros",
+    repeat=1,
+    profile=False,
+):
+    """Run a compiled graph runtime module locally or remotely with
+    optional input values.
+
+    If input tensors are not specified explicitly, they can be filled
+    with zeroes, ones or random data.
+
+    Parameters
+    ----------
+    module_file : str
+        The path to the module file (a .tar file).
+    hostname : str
+        The hostname of the target device on which to run.
+    port : int, optional
+        The port of the target device on which to run.
+    rpc_key : str, optional
+        The tracker key of the target device. If this is set, it
+        will be assumed that remote points to a tracker.
+    device: str, optional
+        the device (e.g. "cpu" or "gpu") to be targeted by the RPC
+        session, local or remote).
+    inputs : dict, optional
+        A dictionary of {input_name: np.array} storing the input
+        tensors to the network. If this is not specified, data will
+        be automatically generated for the input tensors.
+    fill_mode : str, optional
+        The fill-mode to use when generating data for input tensors.
+        Valid options are "zeros", "ones" and "random".
+        Defaults to "zeros".
+    repeat : int, optional
+        How many times to repeat the run.
+    profile : bool
+        Whether to profile the run with the debug runtime.
+
+    Returns
+    -------
+    outputs : dict
+        a dictionary with output tensors, generated by the module
+    times : list of str
+        execution times generated by the time evaluator
+    """
+    if not inputs:
+        inputs = {}
+
+    with tempfile.TemporaryDirectory() as tmp_dir:
+        logger.debug("extracting module file %s", module_file)
+        t = tarfile.open(module_file)
+        t.extractall(tmp_dir)
+        graph = open(os.path.join(tmp_dir, "mod.json")).read()
+        params = bytearray(open(os.path.join(tmp_dir, "mod.params"), "rb").read())
+
+        if hostname:
+            # Remote RPC
+            if rpc_key:
+                logger.debug("running on remote RPC tracker with key %s", rpc_key)
+                remote = request_remote(rpc_key, hostname, port, timeout=1000)
+            else:
+                logger.debug("running on remote RPC with no key")
+                remote = rpc.connect(hostname, port)
+        else:
+            # Local
+            logger.debug("running a local session")
+            remote = rpc.LocalSession()
```

Comment:
Given that `remote` could be a local session, we should use a more general name, such as `session`.

Reply:
Fixed